aten/src/ATen/core/List.h

```diff
@@ -392,12 +391,21 @@ class List final {
   void resize(size_type count, const T& value) const;
 
   /**
-   * Compares two lists for equality. Two lists are equal if they have the
-   * same number of elements and for each list position the elements at
-   * that position are equal.
+   * Value equality comparison. This function implements Python-like semantics for
+   * equality: two lists with the same identity (e.g. same pointer) trivially
+   * compare equal, otherwise each element is compared for equality.
    */
-  friend bool list_is_equal<T>(const List& lhs, const List& rhs);
+  template <class T_>
+  friend bool operator==(const List<T_>& lhs, const List<T_>& rhs);
 
+  template <class T_>
+  friend bool operator!=(const List<T_>& lhs, const List<T_>& rhs);
+
+  /**
+   * Identity comparison. Returns true if and only if `rhs` represents the same
+   * List object as `this`.
+   */
+  bool is(const List<T>& rhs) const;
```

Comment:
Just a style thing, but noting that here you're giving the type argument explicitly for the param, but not in `Dict::is`.

...also (for both List and Dict), I'm assuming wanted `is` defined only within the specialization, right?